services/controller/src/plz/controller/results/local.py

```diff
@@ -38,7 +44,9 @@ def publish(self,
 
             write_bytes(paths.logs, logs)
             write_bytes(paths.output, output_tarball)
-
+            with open(paths.metadata, 'w') as metadata_file:
+                metadata_file.write(json.dumps(
+                    self.compile_metadata(execution_id, finish_timestamp)))
```

Comment:
You can write in a streaming fashion with `json.dump(…, metadata_file)`.

Reply:
Me being sleepy: the movie